src/GoodList.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class GoodsList extends React.Component {
+  state = {
+    goodsLength: 1,
+    goods: [...this.props.goods],
+  }
+
+  reverse = () => {
+    this.setState(prevState => ({
+      goods: [...prevState.goods]
+        .reverse(),
+    }));
+  }
+
+  sortAlphabetically = () => {
+    this.setState(prevState => ({
+      goods: [...prevState.goods]
+        .sort(),
+    }));
+  }
+
+  sortByLength = () => {
+    this.setState(prevState => ({
+      goods: [...prevState.goods]
+        .sort((a, b) => (a.length - b.length)),
+    }));
+  }
+
+  reset = () => {
+    this.setState({
+      goods: [...this.props.goods],
+      goodsLength: 1,
+    });
+  }
+
+  select = (event) => {
+    const { value } = event.target;
+
+    this.setState(prevState => ({
+      goodsLength: value,
+      goods: this.props.goods
+        .filter(good => (good.length >= value)),
+    }));
+  }
+
+  render() {
+    return (
+      <>
+        <ul>
+          {this.state.goods.map(good => (
+            <li key={good}>{good}</li>
+          ))}
+        </ul>
+
+        <div className="control">
+          <button type="button" onClick={this.reverse}>
+            Reverse
+          </button>
+          <button type="button" onClick={this.sortAlphabetically}>
+            Sort A-Z
+          </button>
+          <button type="button" onClick={this.reset}>
+            Reset
+          </button>
+          <button type="button" onClick={this.sortByLength}>
+            Sort by size
+          </button>
+
+          <select
+            onInput={this.select}
+            defaultValue={this.state.goodsLength}
+          >
+            {[...Array(10).keys()].map((item, index) => (
```

Comment:
create this array outside of render, you don't need to create a new one on each render